sda-auth/dev-server/docker-compose.yml

```
@@ -43,6 +43,19 @@ services:
       - CEGA_USERS_USER=dummy
     ports:
       - 8443:8443
+  keygen:
+    image: golang:alpine3.16
+    container_name: keygen
+    command:
+      - "/bin/sh"
+      - "-c"
+      - if [ ! -f "/out/c4gh_key.sec.pem" ]; then git --help >/dev/null 2>&1; [[ "$$?" != "0" ]] && apk add git;
+        [ ! -d "crypt4gh" ] && git clone https://github.com/neicnordic/crypt4gh.git;
+        cd crypt4gh;
+        [ ! -f "crypt4gh" ] && go build .;
```

```suggestion
      - if [ ! -f "/out/c4gh_key.sec.pem" ]; then wget -qO- "https://github.com/neicnordic/crypt4gh/releases/download/v1.8.2/crypt4gh_linux_x86_64.tar.gz" | tar zxf -;
```
I think it is not necessary to use git clone and then compile but instead you can get the binary by using wget

---

Right! This was a blind copy-paste from another repo, we should improve it there too.

---

Given that the crypt4gh binary is published for linux, mac and windows there is zero reason for a runtime compile step.

---

Just check the architecture before and then wget it:
`ARCH=$(uname -sm | sed 's/ /_/' | tr '[:upper:]' '[:lower:]')`
`wget -qO- "https://github.com/neicnordic/crypt4gh/releases/latest/download/crypt4gh_$ARCH.tar.gz" | tar zxf -`
Then it will be independent of the architecture

---

I used a fixed link since we're on alpine.